docs/development/plugins/functionality.md

```diff
@@ -17,7 +17,7 @@ The main ones are:
 * Registry: The plugin registry, used to register functionality
 * Headlamp: To register plugins
 * CommonComponents: React components commonly used in the Headlamp UI
-
+* Notification: This module contains two exported members one is Notification a class which can be used to prepare notifications that are accepted by headlamp and the other one is setNotificationsInStore use this to fill the headlamp redux store with the desired notification.
```

Comment:
Do not remove the newline. Add this before (and also needs a grammar check).
```suggestion
* Notification: This module contains two exported members. One is `Notification`, a class which can be used to prepare notifications that are accepted by Headlamp, and the other is `setNotificationsInStore` which is used to fill the Headlamp redux store with the desired notification.
```

It's unclear to me, reading this, what exactly setNotificationsInStore does or why would we need to call it.

Reply:
Probably an example plugin will help explaining what it does.